lab_5_scraper/scraper.py

```diff
@@ -129,11 +241,42 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
-
+        links = article_bs.find_all('a', class_="news_pic")
+        for link in links:
+            href = link.get('href')
+            if not href:
+                continue
+
+            if href.startswith(('http://', 'https://')):
+                if href not in self.urls:
+                    return href
+                continue
+            seed_urls = self._config.get_seed_urls()
+            if not seed_urls:
+                continue
+
+            base_url = seed_urls[0]
+            if '://' in base_url:
+                base_url = base_url.split('/')[0] + '//' + base_url.split('/')[2]
+
+            full_url = urljoin(base_url, href)
+            if full_url not in self.urls:
+                return full_url
+
+        return ''
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        for seedurl in self.get_search_urls():
+            response = make_request(seedurl, self._config)
+            if not response.ok:
+                continue
+            soup = BeautifulSoup(response.text, 'lxml')
+            url = self._extract_url(soup)
```

Comment:
change the logic of this process, you should get soup, get divs and give them to `_extract_url`